src/vg/civcraft/mc/bettershards/misc/TeleportInfo.java

```diff
@@ -42,6 +44,32 @@ public String toString() {
 		return builder.toString();
 	}
 	
+	@Override
+	public boolean equals(Object obj) {
+		if (obj == null) {
+			return false;
+		}
+
+		if (getClass() != obj.getClass()) {
+			return false;
+		}
+
+		final TeleportInfo other = (TeleportInfo) obj;
+		
+		try {
+			if (!(world.equalsIgnoreCase(other.getWorld())
+					|| (Bukkit.getWorld(UUID.fromString(world)).getName().equalsIgnoreCase(other.getWorld())
```

Comment:
This seems like a big NPE source, especially considering that often TeleportInfos are handled, which are not on this server.